src/utils/mtev_lfu.c

```diff
@@ -269,16 +280,19 @@ mtev_lfu_put(mtev_lfu_t *lfu, const char *key, size_t key_len, void *val)
     return mtev_false;
   }
 
-  unsigned long hash = CK_HS_HASH(&lfu->hash, lfu_entry_hash, key);
+  struct lfu_key *tempkey = alloca(sizeof(struct lfu_key) + key_len);
+  tempkey->key_len = key_len;
+  memcpy(tempkey->key, key, key_len);
```

Comment:
note: you could avoid the `alloca` and `memcpy` if you changed the key type to include a pointer, as well as the flexible array element at the end... Lookups / compares would use the pointer field, and when stored in the heap, the pointer would point to the flexible array element.